lm75bd/lm75bd.c

```diff
@@ -25,9 +25,33 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
   return ERR_CODE_SUCCESS;
 }
 
+/**
+ * @brief Converts data from LM75B temperature sensor to Celcius
+ * 
+ * @param devAddr - I2C device address of sensor
+ * @param temp - Pointer to store converted temperature value in Celsius
+ * @return ERR_CODE_SUCCESS if reading and conversion succeed
+ */
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
-  
+
+  // Stores temperature register value
+  uint8_t sendBuffer = 0x0U;
+  uint8_t buffer[2];
+  uint16_t temperature = 0;
+
+  // Use pointer register to select internal temperature register
+  i2cSendTo(devAddr, &sendBuffer, 1U);
+
+  // Get data on temperature sensor
+  i2cReceiveFrom(devAddr, buffer, 2U);
+
+  // Convert buffer to one binary integer
+  temperature = ((buffer[0] << 8) | buffer[1]) >> 5;
+
+  if((temperature >> 10) == 0) *temp = (float)(temperature * 0.125);
+  // 0xFFFF deletes extra ones from flipped temperature data
+  else *temp = -(((~temperature) & (0xFFFF >> 5)) + 1)*0.125;
```

Comment:
If temperature is a int16_t then you don't need to do these lines just `*temp=(float)(temperature * 0.125);` will do